compose/project.py

```diff
@@ -335,7 +335,10 @@ def _get_convergence_plans(self, services, strategy):
         return plans
 
     def pull(self, service_names=None, ignore_pull_failures=False):
-        for service in self.get_services(service_names, include_deps=True):
+        pull_dependencies = True
+        if service_names:
+            pull_dependencies = False
+        for service in self.get_services(service_names, include_deps=pull_dependencies):
```

Comment:
Let's just do `include_deps=False` here. If there are no services_names it's already going to pull everything, so the behaviour should be the same.